src/main/java/fi/fmi/avi/converter/tac/swx/SWXTACParser.java

```diff
@@ -368,12 +347,72 @@ protected SpaceWeatherAdvisoryAnalysis processAnalysis(final Lexeme lexeme, fina
         return regionList;
     }
 
+    private Geometry buildMultiPolygon(final double minLatitude, final double minLongitude, final double maxLatitude, final double maxLongitude) {
+        if (minLongitude >= maxLongitude && (Math.abs(minLongitude) != 180d && Math.abs(maxLongitude) != 180d)) {
+            List<List<Double>> polygons = new ArrayList<>();
+             if(Math.abs(minLongitude) == 0 && Math.abs(maxLongitude) == 0) {
+                 polygons.add(createPolygon(minLatitude, 0d, maxLatitude, 180d));
+                 polygons.add(createPolygon(minLatitude, -180d, maxLatitude, 0d));
+             } else {
+                 polygons.add(createPolygon(minLatitude, minLongitude, maxLatitude, 180d));
+                 polygons.add(createPolygon(minLatitude, -180d, maxLatitude, maxLongitude));
+             }
+
+
+            return MultiPolygonGeometryImpl.builder().addAllExteriorRingPositions(polygons).setCrs(CoordinateReferenceSystemImpl.wgs84()).build();
```

Comment:
If I'm not wrong, this might be cleaner as

```
return MultiPolygonGeometryImpl.builder()//
        .setCrs(CoordinateReferenceSystemImpl.wgs84())//
        .addExteriorRingPositions(createPolygon(minLatitude, minLongitude, maxLatitude, 180d))//
        .addExteriorRingPositions(createPolygon(minLatitude, -180d, maxLatitude, maxLongitude))//
        .build();
```

Reply:
I made a similar change to what is suggested